src/FAudio.c

```diff
@@ -456,12 +456,12 @@ uint32_t FAudio_CreateSourceVoice(
 	}
 
 	/* Sample Storage */
-	(*ppSourceVoice)->src.decodeSamples = (uint32_t) FAudio_ceil(
-		audio->updateSize *
+	(*ppSourceVoice)->src.decodeSamples = (uint32_t) (FAudio_ceil(
+		(double) audio->updateSize *
 		(double) MaxFrequencyRatio *
 		(double) (*ppSourceVoice)->src.format->nSamplesPerSec /
 		(double) audio->master->master.inputSampleRate
-	) + EXTRA_DECODE_PADDING * (*ppSourceVoice)->src.format->nChannels;
+	) + EXTRA_DECODE_PADDING * (*ppSourceVoice)->src.format->nChannels);
```

Comment:
Oh also! This added paren should be at the end of FAudio_ceil() and not the whole statement, since we're trying to round up for fractions of a sample and _then_ do addition/multiplication on top of that safety padding.

Reply:
fixed